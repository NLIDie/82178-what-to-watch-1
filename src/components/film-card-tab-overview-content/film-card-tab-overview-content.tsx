import * as React from "react";

// Utils
import {getRepresentationRating} from "_Utils/get-representation-rating";

// Types
import {TProps} from "./film-card-tab-overview-content.types";

export function FilmCardTabOverviewContent(props: TProps): JSX.Element {
  const {rating, scoresCount, description, director, starring} = props;

  return (
    <React.Fragment>
      <div className="movie-rating">
        <div className="movie-rating__score">{rating}</div>

        <p className="movie-rating__meta">
          <span className="movie-rating__level">
            {getRepresentationRating(rating)}
          </span>
          <span className="movie-rating__count">{scoresCount} ratings</span>
        </p>
      </div>

      <div className="movie-card__text">
        <p>{description}</p>

        <p className="movie-card__director">
          <strong>Director: {director}</strong>
        </p>

        <p className="movie-card__starring">
          <strong>Starring: {starring.join(`, `)}</strong>
        </p>
      </div>
    </React.Fragment>
  );
}
