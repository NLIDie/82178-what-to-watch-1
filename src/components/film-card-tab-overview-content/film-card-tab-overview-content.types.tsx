// Types
import { TFilm } from "_Types/film";

export type TProps = {
  rating: TFilm['rating'];
  scoresCount: TFilm['scoresCount'];
  description: TFilm['description'];
  director: TFilm['director'];
  starring: TFilm['starring'];
}
