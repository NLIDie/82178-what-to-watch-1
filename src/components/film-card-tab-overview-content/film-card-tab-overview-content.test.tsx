import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {FilmCardTabOverviewContent} from "./film-card-tab-overview-content";

describe(`Components/FilmCardTabOverviewContent`, () => {
  it(`Correct renderer the component`, () => {
    const tree = create(
        <FilmCardTabOverviewContent
          director="director"
          starring={[`star1`, `star2`]}
          description="description"
          rating={8.2}
          scoresCount={40000}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
