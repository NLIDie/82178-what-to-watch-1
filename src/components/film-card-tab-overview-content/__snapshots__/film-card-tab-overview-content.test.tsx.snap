// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FilmCardTabOverviewContent Correct renderer the component 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.2
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        very good
      </span>
      <span
        className="movie-rating__count"
      >
        40000
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      description
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        director
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        star1, star2
      </strong>
    </p>
  </div>,
]
`;
